frontend/src/metabase-lib/lib/Dimension.ts

```diff
@@ -1245,12 +1245,12 @@ export class ExpressionDimension extends Dimension {
     if (!type.startsWith("type/")) {
       base_type = "type/Float"; // fallback
 
-      switch (type) {
-        case MONOTYPE.String:
+      switch (type.toLowerCase()) {
```

Comment:
Actually all type information is intended to be case sensitive (at least, that's the expectation throughout the codebase and deviating that might introduce latent issue).
Is there a reason for `toLowerCase()` here? Was there a case where the case-sensitive comparison failed?

Reply:
It was suggest by @daltojohnso in previous comment but if it is case-sensitive, let's scratch that.

Reply:
That's weird @daltojohnso. If there's a simple test case, send it me way.
We have decent unit tests for the type inference, including for `concat`: https://github.com/metabase/metabase/blob/00b6d1c8eb11b7e1d3017cf2db0acd2de99c9e38/frontend/test/metabase/lib/expressions/typeinferencer.unit.spec.js#L86